include/seqan3/range/detail/inherited_iterator_base.hpp

```diff
@@ -138,7 +162,7 @@ class inherited_iterator_base : public base_t
     }
 
     //!\brief Post-increment, return previous iterator state.
-    constexpr derived_t operator++(int) noexcept(noexcept(++base_t{}))
+    constexpr derived_t operator++(int) noexcept(noexcept(++std::declval<base_t &>()))
```

Comment:
```suggestion
    constexpr derived_t operator++(int) noexcept(noexcept(std::declval<base_t &>()++))
```

Reply:
Note that post increment involves a copy of the iterator, which might throw. I think for this class it is enough to check if the post increment of the base class is noexcept because you store at most a pointer in this class. So copying this class will be noexcept if the base class is.

Reply:
But, if the base_t is an input iterator Post increment might not copy and thus noexcept would still be wrong here if the copy constructor of this input iterator might throw :thinking: maybe you still need to check that also this is noexcept copyable

Reply:
I am checking now if it is noexcept...copyable